pkg/services/publicdashboards/service/service.go

```diff
@@ -180,25 +189,67 @@ func (pd *PublicDashboardServiceImpl) updatePublicDashboardConfig(ctx context.Co
 	return dto.PublicDashboard.Uid, pd.store.UpdatePublicDashboardConfig(ctx, cmd)
 }
 
-// BuildPublicDashboardMetricRequest merges public dashboard parameters with
-// dashboard and returns a metrics request to be sent to query backend
-func (pd *PublicDashboardServiceImpl) BuildPublicDashboardMetricRequest(ctx context.Context, dashboard *models.Dashboard, publicDashboard *PublicDashboard, panelId int64) (dtos.MetricRequest, error) {
+func (pd *PublicDashboardServiceImpl) GetQueryDataResponse(ctx context.Context, skipCache bool, reqDTO *PublicDashboardQueryDTO, panelId int64, accessToken string) (*backend.QueryDataResponse, error) {
+	if err := validation.ValidateQueryPublicDashboardRequest(reqDTO); err != nil {
+		return nil, ErrPublicDashboardBadRequest
+	}
+
+	publicDashboard, dashboard, err := pd.GetPublicDashboard(ctx, accessToken)
```

Comment:
service.GetPublicDashboard checks for IsEnabled. However, I like that it's explicit on this method. I'm a little torn between leaving as is vs switching to call store.GetPublicDashboard and checking for nil/IsEnabled directly in this method.

Reply:
I won't put that in the store since it is business logic. We can keep it inside `GetPublicDashbaord` I will remove it from this method.